### PR TITLE
New extent

### PR DESCRIPTION
**Pull request by amaurea**
New version of extent and extent_subgrid, and a new extent_cyl that's fast and exact but only works for cylindrical projections. The nex extent_subgrid is as general as the previous one, but actually works - it's about 0.1% accurate for a full-sky cylindrical projection with the default settings (though normally one would use extent_cyl in this case), and unlike the previous version keeps increasing in accuracy as nsub is increased. The main extent() automatically selects the appropriate extent implementation based on the wcs that's passed in - the "method" argument is ignored. This could be changed if necessary.

### CODE REVIEW

**Comment by mhasself** [comment]
This PR does not include the tests I added to test_geom.py.  Tests Are Good.

**Comment by msyriac** [comment]
I will merge in Matthews PR, so that should bring in the tests. Note that this PR changes Fourier frequencies for patches far from the equator significantly. For deep56 and boss, this causes a change in a typical binned FFT based power spectrum at the sub percent level and 2 percent level respectively. For deep8, the change is at the 20% level, but one should have known FFTs would be wrong for such a patch.

**Comment by msyriac** [comment]
Other things this PR will probably affect: k-space filtering, flat sky lensing.

File: pixell/enmap.py [hunk 1]
```diff
@@ -565,20 +565,22 @@ def rand_gauss_iso_harm(shape, wcs, cov, pixel_units=False):
...
-		return extent_subgrid(shape, wcs, nsub=nsub, signed=signed)
-	else:
-		raise ValueError("Unrecognized extent method '%s'" % method)
+def extent(shape, wcs, nsub=None, signed=False, method="auto"):
+	"""Returns the area of a patch with the given shape
+	and wcs, in steradians."""
+	if method == "auto":
+		if   wcsutils.is_plain(wcs): method = "intermediate"
+		elif wcsutils.is_cyl(wcs):   method = "cylindrical"
+		else:                        method = "contour"
```
Comment (msyriac) [inline pixell/enmap.py]: This line doesn't seem needed.

**Comment by msyriac** [comment]
Some updated info for deep8:
The ell_x Fourier frequencies change by 0.8%, and ell_y remain the same.
For a given 2D -> 1D ell binning specification, this means the annuli shift a little bit. For an underlying noisy 2D PS like the one I was playing with, noise fluctuations can move across bins between the two branches and cause the max 20% difference I was seeing. The difference is of course smaller the wider the bins. I don't think this is anything to worry about.

**Comment by msyriac** [comment]
@mhasself can you confirm that Sigurd has merged in your contributions fully? I've updated the tests and am ready to merge this in; though we should immediately follow it up with a minor version bump and add notes saying that extent, area and Fourier frequencies have changed with this release.

**Comment by mhasself** [comment]
> @mhasself can you confirm that Sigurd has merged in your contributions fully?

Yes, I'm happy with it.

**Pull request merged**